tests/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::Result;
+use common::{random_payload, send_get, send_post, OrderCreatedResponse, OrderStatusResponse};
+use std::time;
+mod common;
+
+const RETRY_TIMES: u32 = 5;
+const INTERVAL_IN_SECS: u64 = 5;
+const ORDER_STATUS_CREATED: &str= "created";
+const ORDER_STATUS_FULFILLED: &str = "fulfilled";
+
+#[tokio::test]
+async fn create_order_test() -> Result<()> {
+    let mut i = 0;
+    let host_port = 8002;
+    let base_url = format!("http://localhost:{}", host_port);
```

Comment:
nit: possibly can use [url crate](https://docs.rs/url/latest/url/) for build/validate urls.

Reply:
I don´t see a real benefit here ;-)